attachments/simple_engine/mesh_component.h

```diff
@@ -0,0 +1,461 @@
+    [[nodiscard]] const std::string& GetBaseColorTexturePath() const { return baseColorTexturePath; }
+    [[nodiscard]] const std::string& GetNormalTexturePath() const { return normalTexturePath; }
+    [[nodiscard]] const std::string& GetMetallicRoughnessTexturePath() const { return metallicRoughnessTexturePath; }
+    [[nodiscard]] const std::string& GetOcclusionTexturePath() const { return occlusionTexturePath; }
+    [[nodiscard]] const std::string& GetEmissiveTexturePath() const { return emissiveTexturePath; }
+
+    /**
+     * @brief Create a simple sphere mesh.
+     * @param radius The radius of the sphere.
+     * @param color The color of the sphere.
+     * @param segments The number of segments (resolution).
+     */
+    void CreateSphere(float radius = 1.0f, const glm::vec3& color = glm::vec3(1.0f), int segments = 16);
+
+    /**
+     * @brief Load mesh data from a Model.
+     * @param model Pointer to the model to load from.
+     */
+    void LoadFromModel(const class Model* model);
+
+    // Instancing methods
+
+    /**
+     * @brief Add an instance with the given transform matrix.
+     * @param transform The transform matrix for this instance.
+     * @param materialIndex The material index for this instance (default: 0).
+     */
+    void AddInstance(const glm::mat4& transform, uint32_t materialIndex = 0) {
+        instances.emplace_back(transform, materialIndex);
+        isInstanced = instances.size() > 1;
+    }
+
+    /**
+     * @brief Set all instances at once.
+     * @param newInstances Vector of instance data.
+     */
+    void SetInstances(const std::vector<InstanceData>& newInstances) {
+        instances = newInstances;
+        isInstanced = instances.size() > 1;
+    }
+
+    /**
+     * @brief Get all instance data.
+     * @return Reference to the instances vector.
+     */
+    [[nodiscard]] const std::vector<InstanceData>& GetInstances() const {
+        return instances;
+    }
+
+    /**
+     * @brief Get the number of instances.
+     * @return Number of instances (0 if not instanced, >= 1 if instanced).
+     */
+    [[nodiscard]] size_t GetInstanceCount() const {
+        return instances.size();
+    }
+
+    /**
+     * @brief Check if this mesh uses instancing.
+     * @return True if instanced (more than 1 instance), false otherwise.
+     */
+    [[nodiscard]] bool IsInstanced() const {
+        return isInstanced;
+    }
+
+    /**
+     * @brief Clear all instances and disable instancing.
+     */
+    void ClearInstances() {
+        instances.clear();
+        isInstanced = false;
+    }
+
+    /**
+     * @brief Update a specific instance's transform.
+     * @param index The index of the instance to update.
+     * @param transform The new transform matrix.
+     * @param materialIndex The new material index (optional).
+     */
+    void UpdateInstance(size_t index, const glm::mat4& transform, uint32_t materialIndex = 0) {
+        if (index < instances.size()) {
```

Comment:
Should there be some error handling if `instances.size() <= index`?
Returning false, or an assertion? Something that tells the programmer that he is wrong.